File: app/src/main/java/com/k4dima/chat/core/model/Message.java
package com.k4dima.chat.core.model;

import android.arch.persistence.room.Embedded;
import android.arch.persistence.room.Entity;
import android.arch.persistence.room.PrimaryKey;
import android.arch.persistence.room.TypeConverters;

import com.k4dima.chat.core.db.DateConverter;
import com.stfalcon.chatkit.commons.models.IMessage;

import java.util.Date;

@Entity
public class Message implements IMessage {
    @PrimaryKey(autoGenerate = true)
    public long id;
    private String text;
    @TypeConverters({DateConverter.class})
    private Date createdAt;
    @Embedded(prefix = "user")
    private User user;

    public Message(User user, String text) {
        this(user, text, new Date());
    }

    private Message(User user, String text, Date createdAt) {
        this.text = text;
        this.user = user;
        this.createdAt = createdAt;
    }

    @Override
    public String getId() {
        return "" + id;
    }

    public void setId(String id) {
        this.id = Integer.parseInt(id);
    }

    @Override
    public String getText() {
        return text;
    }

    public void setText(String text) {
        this.text = text;
    }

    @Override
    public Date getCreatedAt() {
        return createdAt;
    }

    public void setCreatedAt(Date createdAt) {
        this.createdAt = createdAt;
    }

    @Override
    public User getUser() {
        return this.user;
    }
}

File: app/src/main/java/com/k4dima/chat/core/model/Message.kt
package com.k4dima.chat.core.model

import androidx.room.Embedded
import androidx.room.Entity
import androidx.room.PrimaryKey
import androidx.room.TypeConverters
import com.k4dima.chat.core.db.DateConverter
import com.stfalcon.chatkit.commons.models.IMessage
import java.util.*

@Entity
class Message private constructor(
        @Embedded(prefix = "user")
        private val user: User,
        private val text: String,
        @field:TypeConverters(DateConverter::class)
        private var createdAt: Date
) : IMessage {
    @PrimaryKey(autoGenerate = true)
    var id: Long = 0

    constructor(user: User, text: String) : this(user, text, Date())

    override fun getId(): String {
        return "" + id
    }

    override fun getText(): String {
        return text
    }

    override fun getCreatedAt(): Date {
        return createdAt
    }

    fun setCreatedAt(createdAt: Date) {
        this.createdAt = createdAt
    }

    override fun getUser(): User {
        return this.user
    }
}